lib/DeleteItemBuilder.js

```diff
@@ -13,13 +13,15 @@ function DeleteItemBuilder(options) {
 require('util').inherits(DeleteItemBuilder, Builder)
 
 DeleteItemBuilder.prototype.execute = function () {
-  var queryData = new DynamoRequest(this.getOptions())
+  var req = new DynamoRequest(this.getOptions())
     .setTable(this._tablePrefix, this._table)
     .returnConsumedCapacity()
     .setHashKey(this._hashKey, true)
-    .setRangeKey(this._rangeKey, true)
     .setExpected(this._conditions)
-    .build()
+    
+  if (this._rangeKey) req.setRangeKey(this._rangeKey, true)
```

Comment:
would it make sense for DynamoRequest to handle this internally? I could go either way on this.